code/modules/surgery/helpers.dm

```diff
@@ -22,11 +22,15 @@
 				for(var/datum/surgery/S in all_surgeries)
 					if(!S.possible_locs.Find(selected_zone))
 						continue
-					if(affecting && S.requires_organic_bodypart && affecting.status == ORGAN_ROBOTIC)
+					if(affecting)
+						if(S.requires_missing_bodypart)
+							continue
+						if(S.requires_organic_bodypart && affecting.status == ORGAN_ROBOTIC)
+							continue
+					else if(H && !S.requires_missing_bodypart) //human with no limb in surgery zone when we need a limb

```

Comment:
I don't think so. the var name isn't great so if you have better names I'm all ears.